### PR TITLE
:warning: remove deprecated go/v2 which provides legacy layout - KB CLI 2x < 3.0.0

### PR DESCRIPTION
**Pull request by camilamacedo86**
## Description

Remove legacy layout kept to ensure backwards compatibility with KB CLI 2x.
Remove the old layout for the Project file used by it - Project config version 2.

## Motivation

We are unable to keep this plugin maintained and it does not work with the plugin ecosystem and/or the latest versions. More info: https://github.com/kubernetes-sigs/kubebuilder/discussions/3003

Closes: https://github.com/kubernetes-sigs/kubebuilder/issues/3002

/hold 
Until we are able to move forward. So, far it is planned for when we bump k8s 1.26

### CODE REVIEW

File: pkg/plugin/util/testdata/exampleFile.txt [hunk 1]
```diff
@@ -1 +1 @@
-exampleTarget
\ No newline at end of file
+exampleTargetexampleCodeexampleCode
```
Comment (everettraven) [inline pkg/plugin/util/testdata/exampleFile.txt:1]: I think we need to make sure this change doesn't get committed right? I *think* there is a test that fails if this gets committed as is

**Review by everettraven — APPROVED** [review]
Overall LGTM. Only one question (probably non-blocking):

**Comment by camilamacedo86** [comment]
/hold 

Until we decide to move forward within

**Comment by camilamacedo86** [comment]
We can close this one in favor to do that within a 4.x release.
Also, we can use this as example over what should be done.

**Pull request closed without merging**